streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -538,6 +544,8 @@ protected Map<String, Integer> getNumPartitions(final Set<String> topics,
                     tempUnknownTopics.add(topicName);
                     log.debug("The leader of topic {} is not available.\n" +
                         "Error message was: {}", topicName, cause.toString());
+                } else if (cause instanceof TimeoutException) {
+                    throw new RuntimeException();
```

Comment:
We should not throw an exception here, but do what we do inside the existing `catch TimeoutException` block (that we can remove).